website/docs/enterprise/pipelines/pipeline-templates.mdx

```diff
@@ -0,0 +1,278 @@
+      promotion:
+        pull-request:
+            url: ${APPLICATION_REPO_URL}
+            branch: ${APPLICATION_REPO_BRANCH}
+            secretRef:
+              name: ${GIT_CREDENTIALS_SECRET}
+```
+
+#### Git credentials
+For guidance on configuring credentials - see instructions in the [Promoting Applications](../promoting-applications#create-credentials-secret) documentation.
+
+#### Promotion marker to be added to HelmRelease in `Second-Environment`
+A comment would need to be added to the HelmRelease or Kustomization patch where the `spec.chart.spec.version` is defined.
+
+For example, if the values used in the above template were as follows:
+
+```yaml
+RESOURCE_NAME=my-app
+RESOURCE_NAMESPACE=pipeline-01
+```
+
+Then the marker would be:
+
+```yaml
+# {"$promotion": "pipeline-01:my-app:Second-Environment"}
+```
+
+More guidance on adding markers can be found [here](../promoting-applications#add-markers-to-app-manifests).
+
+### Alerts and Providers
+This example shows how you can configure multiple resources in a single template, and simplify creation through common naming strategies. The notification controller is used to communicate update events from the leaf clusters where Applications are deployed, to the management cluster where the Pipeline Controller resides and orchestrates.
+
+For the `Alert`, this template is filtering events to detect when an update has occurred. Depending on your use case, you could use different filtering.
+
+For the `Provider`, this template uses authenticated (HMAC) communication to the promotion endpoint, where a secret will need to be present on both the management cluster and leaf cluster(s). For simplicity, a `generic` provider could be used instead - which would not require the secret.
+
+```yaml
+apiVersion: templates.weave.works/v1alpha1
+kind: GitOpsTemplate
+metadata:
+  name: pipeline-notification-resources
+  namespace: flux-system
+  labels:
+    weave.works/template-type: application # These are generic Flux resources rather than Pipeline-specific
+spec:
+  description: Creates flux notification controller resources for a cluster, required for promoting applications via pipelines.
+  params:
+    - name: RESOURCE_NAME 
+      description: Name for the generated objects, should match the target Application (HelmRelease) name.
+    - name: RESOURCE_NAMESPACE
+      description: Namespace for the generated objects, should match the target Application (HelmRelease) namespace.
+    - name: PROMOTION_HOST
+      description: Host for the promotion webhook on the management cluster, i.e. "promotions.example.org"
+    - name: ENV_NAME
+      description: Environment the cluster is a part of within a pipeline.
+  resourcetemplates:
+  - apiVersion: notification.toolkit.fluxcd.io/v1beta1
+    kind: Provider
+    metadata:
+      name: ${RESOURCE_NAME}
+      namespace: ${RESOURCE_NAMESPACE}
+    spec:
+      address: http://${PROMOTION_HOST}/promotion/${APP_NAME}/${ENV_NAME}
+      type: generic
```

Comment:
```suggestion
      type: generic-hmac
```
if using `generic-hmac` then we should include the `secretRef` field, currently on appRef

Reply:
added to the provider